### PR TITLE
Python: Reduce the use of mock objects

### PR DESCRIPTION
**Pull request by Fokko**
We use mocks extensively in our Python code, this was before we had certain functionality available, such as a working FileIO.

Instead of using the mocks, we can also use the actual code.

### CODE REVIEW

File: python/pyiceberg/avro/reader.py [hunk 1]
```diff
@@ -104,16 +104,22 @@ def _skip_map_array(decoder: BinaryDecoder, skip_entry: Callable[[], None]) -> N
         block_count = decoder.read_int()
 
 
-@dataclass(frozen=True)
 class AvroStruct(StructProtocol):
-    _data: List[Union[Any, StructProtocol]] = dataclassfield()
+    _data: List[Union[Any, StructProtocol]]
```
Comment (rdblue) [inline python/pyiceberg/avro/reader.py:108]: Is anything here specific to Avro? Or should this be a generic struct in a different place?

**Comment by rdblue** [comment]
Looks good to me! I'm not merging in case you want to move/rename `AvroStruct` since it is really a generic class and not specific to Avro.

**Comment by rdblue** [comment]
Looks like this needs to be rebased.

**Comment by Fokko** [comment]
@rdblue rebased! 👍🏻

**Pull request merged**